packages/csvviewer/src/widget.ts

```diff
@@ -181,16 +166,52 @@ namespace CSVViewer {
   }
 }
 
+export
+class CSVDocumentWidget extends DocumentWidget<CSVViewer> {
+  constructor(options: CSVDocumentWidget.IOptions) {
+    let {content, context, delimiter, ready, ...other} = options;
+    content = content || Private.createContent(context);
+    ready = Promise.all([ready, content.ready]);
+    super({content, context, ready, ...other});
```

Comment:
Did something change with TS? I thought that the `super` call had to be first in a constructor.

Reply:
The super has to be before you use any attributes of the object. Since I'm just creating local variables, it's fine.